test/test_testing.py

```diff
@@ -971,6 +971,27 @@ def test_default_tolerance_selection_mismatching_dtypes(self):
         for fn in assert_close_with_inputs(actual, expected):
             fn(check_dtype=False)
 
+    class UnexpectedException(Exception):
```

Comment:
Comment here explaining why this exception is defined and what the following tests do would be nice